replication_test.go

```diff
@@ -172,6 +173,65 @@ func TestReplicationStartsBeforeCurrentRound(t *testing.T) {
 	}
 }
 
+func TestReplicationFutureFinalizationCertificate(t *testing.T) {
+	// send a block, then simultaneously send a finalization certificate for the block
+	l := testutil.MakeLogger(t, 1)
+	bb := newTestDelayedVerificationBlockBuilder(50 * time.Millisecond)
```

Comment:
the test passes even when i replace this with `newTestControlledBlockBuilder`. 

The 50ms sleep is not something we should rely on... can't we make something more controlled? 

How about doing this and then you can ditch the entire `testDelayedVerificationBlockBuilder` and its friends?

```
func (tb *testBlock) Verify() error {
	if tb.verificationDelay == nil {
		return nil
	}
	
	<- tb.verificationDelay
	
	return nil
}
```

Then all we need is to add:

```
	err = e.HandleMessage(&simplex.Message{
		FinalizationCertificate: &fCert,
	}, nodes[0])
	require.NoError(t, err)

	block.verificationDelay <- struct{}{} // <----

	storedBlock := storage.waitForBlockCommit(0)
```